src/common/string/compare.ts

```diff
@@ -11,3 +11,25 @@ export const compare = (a: string, b: string) => {
 
 export const caseInsensitiveCompare = (a: string, b: string) =>
   compare(a.toLowerCase(), b.toLowerCase());
+
+export const fuzzyCompare = (filter: string, word: string) => {
```

Comment:
```ts
const regex = new RegExp(filter.split("").reduce(function(a,b){ return a+'[^'+b+']*'+b; }));

return regex.test(word);
```

Is something with regex faster for these cases?

Reply:
I'll see if I can benchmark it.

Though the "user-feel" response (for me anyway) is so fast that it seems like readability oughta win out over perf. 

Still, might be good to test and see.

Reply:
Faster is usually better. And this regex isn't actually that hard to understand. Its all wildcards. Also, have to think about people with hundreds of entities

Reply:
Looks like regex is about 67% slower:
![image](https://user-images.githubusercontent.com/1480827/92298395-1e420080-eefd-11ea-8a36-777f617f805a.png)

Reply:
Nice 👍 I wasnt for sure

Reply:
Oh, also I think your regex would fail on first page load, where the filter string is an empty string to start. You'd need to add an initial value after the callback declaration.

Reply:
Yea Im sure it wasnt perfect. It was more "Something like this" :) I have just seen regex for this before so I wasn't sure if there was a better way using it